Bring in HAFSv1 nesting needed changes from FRE-NCtools

BinLiu-NOAA opened the pull request:
## DESCRIPTION OF CHANGES: 
- Bring in HAFS nesting needed changes for fregrid, make_hgrid, and make_solo_mosaic from NOAA-GFDL/FRE-NCtools repository.

## TESTS CONDUCTED: 
All the changes have been tested with the UFS HAFS application.

The grid_gen consistency tests were run. Results documented below.

## DEPENDENCIES:
N/A

## DOCUMENTATION:
N/A

## ISSUE: 
Fixes issue mentioned in #751 

## CONTRIBUTORS: 
FRE-NCtools changes are mainly from https://github.com/NOAA-GFDL/FRE-NCtools. The FRE-NCtools related porting were mainly done by @BinLiu-NOAA and @BijuThomas-NOAA.

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66221
Initial commit to support conversion of atmospheric data from grib2
files

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Initial commit to support conversion of atmospheric data from grib2
files

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66221
Bug fix for improper tracer loop in regridding section

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Bug fix for improper tracer loop in regridding section

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Add a module load command for wgrib2 for chgres_cube.theia

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' of gerrit:UFS_UTILS into feature/chgres_cube_grib2_atmos

BinLiu-NOAA committed:
feature/HAFS: Import the regional_grid from Jim/Duson/Jeff/Gerald.

BinLiu-NOAA committed:
feature/HAFS: Add the corresponding module files for regional_grid.

BinLiu-NOAA committed:
feature/HAFS: Change module files for regional_grid on Theia and Jet.

BinLiu-NOAA committed:
Merge branch 'develop' into feature/HAFS

BinLiu-NOAA committed:
Merge branch 'develop' into feature/HAFS

Conflicts (resolved):
	sorc/build_all_ufs_utils.sh
	sorc/partial_ufs_build.sh
	sorc/ufs_build.cfg
	ush/fv3gfs_driver_grid.sh

BinLiu-NOAA committed:
feature/HAFS: Sync with develop for sorc/global_chgres.fd/read_write.f90.

BinLiu-NOAA committed:
feature/HAFS: Enable building sfc_climo_gen on Jet.

BinLiu-NOAA committed:
feature/HAFS: Enable building regional_grid on Jet.

BinLiu-NOAA committed:
Merge branch 'develop' into feature/HAFS

Conflicts (resolved):
	modulefiles/chgres_cube.jet
	modulefiles/fv3gfs/fre-nctools.jet
	modulefiles/fv3gfs/global_chgres.jet
	modulefiles/fv3gfs/orog.jet
	modulefiles/modulefile.sfc_climo_gen.jet
	sorc/fre-nctools.fd/tools/filter_topo/make.csh_jet
	sorc/fre-nctools.fd/tools/make_hgrid/fre-nctools.mk_jet
	sorc/fre-nctools.fd/tools/shave.fd/build_shave

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Added check for bad namelist inputs for grib2 input. Improved varmap
tables and how they're read. Bug Fixes.

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Minor syntax/bug fixes.

JeffBeck-NOAA committed:
Changes so JPgrid will generate grid indices in the same manner as GFDLgrid

GeorgeGayno-NOAA committed:
feature/chgres_cube_grib2_atmos:  This commit references #66268.

chgres_cube.theia build module file - fix error in load of w3nco
library.

Update makefile to link the wgrib2 api last.  This eliminates
problems reading netcdf files which is the result of esmf and
wgrib2 using different versions of netcdf (?)

Change-Id: Ifee9f0986fb6664122acc5c32a7ece8a5e6a8143

GeorgeGayno-NOAA committed:
feature/chgres_cube_grib2_atmos:  This commit references #66268.

Merge branch 'develop' into feature/chgres_cube_grib2_atmos.

Change-Id: Ie4b94468ff2c46ed95f547c4ac8b07586cb59ec8

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Fixed empty line in GFS varmap table and updated reading of varmap table
to ignore empty lines.

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' of gerrit:UFS_UTILS into feature/chgres_cube_grib2_atmos

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Fix for proper RH->sphum computation.

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
dzdt is now properly converted to sigma coordinates and flipped
vertically for non-native grib2 files

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268 (#8)

dzdt is now properly converted to sigma coordinates and flipped
vertically for non-native grib2 files

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Update build of 'fre-nctools', 'nst_tf_chg', 'nems_util' and
'orog' programs for Hera.  Remove references to Theia.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Update 'emcsfc' and 'sfc_climo_gen' build for Hera.

Add better logic to 'machine-setup.sh' to determine if on Hera.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references issue #1.

Update 'chgres', 'chgres_cube' and 'global_cycle' to build
on Hera.

All programs can now be compiled on Hera.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references issue #1.

Update to intel v15 for chgres on phase 1/2.  This
is required because other machines use newer versions
that require the -qopenmp flag.  v14 only recognized
-openmp.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references issue #1.

Update "link_fixdirs.sh" for Hera.  Remove references to
Theia.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Merge branch 'develop' into feature/hera_port

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Create new grid driver script for Hera.  Remove Theia version.

Update sfc_climo_gen build module to point to George V's copy
of esmf.  Other users could not compile using the other
esmf module.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Point to George V's copy of the nesmio library instead of
the 'official' version.  The nemsio utilities did not
work with the 'official' version - they return bad istat (-41)
when reading.  Either the official version was compiled wrong
or as a newer version (v2.5) it may be incompatible with the
programs.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Dexin compiled nemsio v2.2.3 on Hera and it was successfully
tested with the nemsio utilities.  Switch to v2.2.3 for all
programs.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Baseline the batch submission wrappers per request of
global workflow group.  These are used by the
chgres script.  Previously, they resided in the
glopara home directories on each machine.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Updates to run 'fv3gfs_chgres.sh' on Hera.

GeorgeGayno-NOAA committed:
feature/hera_port  This commit references #1.

Remove unused "module-setup.csh.inc"

Update "ush/fv3gfs_chgres.sh" to point to new location for
'sub_wcoss' scripts.  Remove some commented out code.

Update "module-setup.sh.inc" for Hera.

LarissaReames committed:
chgres_cube_grib2_atmos: This commit references #66268
Fixed a new indexing error in creating the staggered grid coordinates.
Also removed the call to iso2sig when data is in isobaric coordinates.
3d pressure array is now generated from pressure levels read from file.

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' into feature/chgres_cube_grib2_atmos

LarissaReames committed:
Feature/chgres cube grib2 atmos (#13)

* chgres_cube_grib2_atmos: This commit references #66268
dzdt is now properly converted to sigma coordinates and flipped
vertically for non-native grib2 files

* chgres_cube_grib2_atmos: This commit references #66268
Fixed a new indexing error in creating the staggered grid coordinates.
Also removed the call to iso2sig when data is in isobaric coordinates.
3d pressure array is now generated from pressure levels read from file.

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Update location of zero bia file on hera.

BinLiu-NOAA committed:
Merge branch 'develop' into feature/HAFS

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Kate moved the climo files for program "sfc_climo_gen" to the
standard location along with the other fixed files.  Update
"sorc/link_fixdirs.sh" accordingly.

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Add the "mkgfsnemsioctl" utility from the global-workflow
repository.

BinLiu-NOAA committed:
Merge branch 'feature/hera_port' into feature/HAFS

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Merge branch 'develop' into feature/hera_port

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Detect if running on Hera by checking for the existence of
the /scratch1 directory per recommendation of the original
designer.  Previously, hostname was used.  There is
some concern that using 'hostname' will not work
on the compute nodes (although I did not see a problem with
my tests).

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

./ush/fv3gfs_chgres.sh - increase Hera wall time to 3 hours.

BinLiu-NOAA committed:
feature/HAFS: More for Hera porting.
Use newer versions of libraries on Hera.
Modify the method to detect the Hera machine.

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Update serial chgres to use netcdf 4.7.0 and hdf 1.10.5
(on hera) per request of global workflow group.

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Hera build modules - Remove unneeded module loads.  Point to
standard location for libraries instead of George V's home
directory.  Update cycle program to use netcdf 4.7 per
request of global workflow group.

BinLiu-NOAA committed:
feature/HAFS: Load newer version of landsfcutil/2.1.1 and esmf/8.0.0bs48.

BinLiu-NOAA committed:
feature/HAFS: Switch back to load netcdf/4.6.1 and hdf5/10.0.4 on hera.

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Update remaining programs to use netcdf 4.7 per request of
global workflow group.  For esmf, use gwv's copy of version
8 compiled with netcdf 4.7.

GeorgeGayno-NOAA committed:
feature/hera_port:  This commit references #1.

Merge branch 'develop' into feature/hera_port

BinLiu-NOAA committed:
Merge branch 'develop' into chgres_cube_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'hera_port' into chgres_cube_grib2_atmos

GeorgeGayno-NOAA committed:
feature/pole_bug:  This commit references #23.

sfc_climo_gen and chgres_cube programs:  For regrid options
other than conservative and neighbor, change pole method from
NONE to ALLAVG.

GeorgeGayno-NOAA committed:
feature/pole_fix:  This commit references #23.

Merge branch 'develop' into feature/pole_bug

GeorgeGayno-NOAA committed:
feature/pole_bug:  This commit references #23.

Merge branch 'develop' into feature/pole_bug

BinLiu-NOAA committed:
Merge pull request #1 from NOAA-EMC/develop

Sync with the official develop

BinLiu-NOAA committed:
Merge branch 'develop' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'feature/pole_bug' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'develop' into feature/chgres_cube_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'feature/chgres_cube_grib2_atmos' into feature/HAFS_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'develop' into feature/HAFS_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'develop' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'develop' into feature/chgres_cube_grib2_atmos

BinLiu-NOAA committed:
support/HAFS: Add support to generate grids for more model resolutions in
ush/fv3gfs_driver_grid.sh

BinLiu-NOAA committed:
support/HAFS: Enable make_hgrid shift_fac option for uniform grid.

Regarding the center longitude of the uniform global grid, the default center
longitude for tile 1 was hardwired at 10.0W, even though in the source code
instructions there is a commandline option (--shift_fac), which can shift the
center longitude. For example:
In make_hgrid.c:
...
  "   --shift_fac                shift west by 180/shift_fac. Default value is 18.  ",
...

In the source code, this shift_fac option, was hard-wired as 18.0
(corresponding to setting the tile 1 center longtintude to 10.0W).

In this commit, the capability of using this shift_fac option is enabled.
Thus, if one wants to specify the center lon for tile 1 to be 60.0W, he can
specify --shift_fac=3.0 (180./3.0=60, meaning shift 60 degrees westward).

In any case, the default value of shift_fac is still 18.0, thus the default
tile 1 center longitude is still 10.0W. This commit will not change the default
behavior, rather than provide the capability/flexibility to cumstomize the tile
1 center longitude if that is desired.

A note from George Gyno: I think GFDL hardwired it to ensure tall mountains
were located away from the tile edges.

LarissaReames committed:
Merge pull request #1 from hafs-community/feature/chgres_cube_grib2_atmos

Feature/chgres cube grib2 atmos

BinLiu-NOAA committed:
Merge branch 'develop' into feature/chgres_cube_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'feature/chgres_cube_grib2_atmos' into feature/HAFS_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'develop' into support/HAFS

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' of https://github.com/LarissaReames-NOAA/UFS_UTILS into chgres_cube_grib2_atmos

JiliDong-NOAA committed:
Merge remote-tracking branch 'LarissaFork/feature/chgres_cube_grib2_atmos' into feature/chgres_cube_grib2_atmos

JiliDong-NOAA committed:
re-order grib2 isobaric levels

JiliDong-NOAA committed:
keep grib2 CLWMR name consistent with origional

LarissaReames committed:
Replaced namelist option "base_install_dir" with "varmap_file" and
"fixed_files_dir_input_grid". Varmap_file must not contain the full path
to the variable mapping table and fixed_files_dir_input must point to the
directory containing fixed files relevant to input data sources (e.g.,
soil and vegetation data for gfs grib2 files).

Also, removed extraneous code that was previously used for converting
isobaric coordinate date to hybrid coordinates before being interpolated
to the target grid.

This commit references issue #5.

LarissaReames committed:
Merge pull request #3 from JiliDong-NOAA/feature/chgres_cube_grib2_atmos

Feature/chgres cube grib2 atmos: re-order grib2 isobaric levels

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' of https://github.com/LarissaReames-NOAA/UFS_UTILS into chgres_cube_grib2_atmos

LarissaReames committed:
Merge pull request #2 from hafs-community/feature/chgres_cube_grib2_atmos

Feature/chgres cube grib2 atmos

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' of https://github.com/LarissaReames-NOAA/UFS_UTILS into chgres_cube_grib2_atmos

GeorgeGayno-NOAA committed:
Merge pull request #2 from LarissaReames-NOAA/feature/chgres_cube_grib2_atmos

Merge feature/chgres cube grib2 atmos from Larissa's fork to my fork.

GeorgeGayno-NOAA committed:
feature/chgres_cube_grib2_atmos:

This commit references #6.

chgres_cube - process gfs grib2 data on its native isobaric
coordinate instead of converting it to a hybrid coordinate.
Remove routines used for this conversion.

LarissaReames committed:
Merge pull request #4 from GeorgeGayno-NOAA/feature/chgres_cube_grib2_atmos

Feature/chgres cube grib2 atmos

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' of https://github.com/LarissaReames-NOAA/UFS_UTILS into chgres_cube_grib2_atmos

GeorgeGayno-NOAA committed:
feature/chgres_cube_grib2_atmos:

This commit references #6.

Remove logic to process input grib2 data on hybrid levels
(isnative=true).  The next public release will only support
gfs grib 2 data which is on isobaric levels.

General clean up.

LarissaReames committed:
grib2 inventory file for gfs data is now saved in the run directory instead of the input data directory

JiliDong-NOAA committed:
Jet support for chgres_cube_grib2_atmos

LarissaReames committed:
Removed a few tabs in favor of spaces in model_grid.F90.

Fixed bug with nullifcation of presptr before use in input_data.F90. Also, now use TMP instead of UGRD to determine vertical levels. Some files have TMP above the highest UGRD entry.

LarissaReames committed:
Merge pull request #6 from JiliDong-NOAA/feature/chgres_cube_grib2_atmos

Jet support for chgres_cube_grib2_atmos

GeorgeGayno-NOAA committed:
feature/chgres_cube_grib2_atmos:

This commit references #6.

Update chgres_cube build module for Dell.

GeorgeGayno-NOAA committed:
Merge remote-tracking branch 'Larissa/feature/chgres_cube_grib2_atmos' into feature/chgres_cube_grib2_atmos

JiliDong-NOAA committed:
fix sorting to handle isobaric levels with decimal (mb)

JiliDong-NOAA committed:
fix sorting to handle isobaric levels with decimal (mb) cleaning up

LarissaReames committed:
All atmospheric variables are now read in from grib2 day based on grib2
parameter numbers instead of by name.

LarissaReames committed:
Merge pull request #7 from JiliDong-NOAA/feature/chgres_cube_grib2_atmos

fix sorting to handle isobaric levels with decimal (mb)

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' of https://github.com/LarissaReames-NOAA/UFS_UTILS into chgres_cube_grib2_atmos

LarissaReames committed:
Merge branch 'feature/chgres_cube_grib2_atmos' into feature/chgres_cube_grib2_atmos

LarissaReames committed:
Merge pull request #5 from GeorgeGayno-NOAA/feature/chgres_cube_grib2_atmos

feature/chgres_cube_grib2_atmos:

LarissaReames committed:
Bug fix for an extraneous line of code left over in last merge.

LarissaReames committed:
Fix bug in creation of grib2 tracer parameter number string arrays.

LarissaReames committed:
Removed extraneous code from a previous merge.

BinLiu-NOAA committed:
Merge branch 'feature/chgres_cube_grib2_atmos' into feature/HAFS_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'support/HAFS' into feature/HAFS_grib2_atmos

BinLiu-NOAA committed:
feature/HAFS_grib2_atmos: Add a new var_map table (FV3GFSphys_var_map.txt) to deal with FV3GFS grib2 files.

BinLiu-NOAA committed:
Merge branch 'develop' into feature/HAFS_grib2_atmos

BinLiu-NOAA committed:
Merge branch 'feature/HAFS_grib2_atmos' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'develop' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'develop' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'develop' into support/HAFS

GeorgeGayno-NOAA committed:
feature/orion   This commit references #96.

Update 'orog' and 'fre-nctools' builds for orion.

GeorgeGayno-NOAA committed:
feature/orion   This commit references #96.

Update 'global_cycle', 'nst_tf_chg', 'nemsutil', 'emcsfc'
build for orion.

GeorgeGayno-NOAA committed:
feature/orion   This commit references #96.

Update 'global_chgres' build for Orion.

GeorgeGayno-NOAA committed:
feature/orion   This commit references #96.

Update sfc_climo_gen build for Orion.

GeorgeGayno-NOAA committed:
feature/orion  This commit references #96.

Add Orion build module for chgres_cube.

GeorgeGayno-NOAA committed:
feature/orion  This commit references #96.

Update "link_fixdirs.sh" to link to Kate's fixed directories
on Orion.

GeorgeGayno-NOAA committed:
feature/orion  This commit references #96.

Baseline script to create grids - 'grid', 'orog', and 'sfc_climo'
files - on Orion.

GeorgeGayno-NOAA committed:
feature/orion  This commit references #96.

Remove all references to WCOSS Phase 1/2 (tide/gyre),
which is being retired.

GeorgeGayno-NOAA committed:
feature/orion  This commit references #96.

Add Orion driver script for chgres_cube regression test suite.

GeorgeGayno-NOAA committed:
feature/orion   This commit references #96.

Update orion chgres regression test script to use threads
for the 'gfs sigio' test.  Update chgres_cube build
modulefile to use splib v2.0.3.  splib v2.1.0 was not working
with threads - the reg test would time out after 15 minutes.
v2.1.0 is the 'cmake' version.

GeorgeGayno-NOAA committed:
feature/orion  This commit references #96.

Update threading in reg_tests/chgres_cube/driver.orion.sh.

GeorgeGayno-NOAA committed:
feature/orion   This commit references #96.

Update sorc/link_fixdirs.sh to point to Kate's
new directories.

GeorgeGayno-NOAA committed:
feature/orion   This commit references #96.

Merge branch 'develop' into feature/orion

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

GeorgeGayno-NOAA committed:
feature/orion  This commit references #96.

Increase stack size for running sigio chgres_cube test
on Orion.

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/GG/feature/orion' into feature/port_to_orion

Conflicts:
	sorc/build_orog.sh
	sorc/fre-nctools.fd/tools/shave.fd/build_shave

BinLiu-NOAA committed:
Enable build regional_grid on Orion.

BinLiu-NOAA committed:
Use the netcdfp/4.7.4 module which is used by ufs-weather-model on Orion.

wramstrom committed:
Added multiple static nest functionality

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/origin/support/HAFS' into feature/multi_nests

BinLiu-NOAA committed:
Changes needed for the Jet directory renaming from /lfs3/projects to /lfs3/HFIP.

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Clean up the support/HAFS branch so that it is closer to the develop branch.

BinLiu-NOAA committed:
module use /usrx/local/dev/modulefiles prior to module load cmake/3.16.2 in
modulefiles/build.wcoss_cray. (From Biju)

BinLiu-NOAA committed:
Merge branch 'support/HAFS' into feature/multi_nests

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'support/HAFS' into feature/multi_nests

BinLiu-NOAA committed:
Load netcdfp/4.7.4 and esmflocal/8.1.0.19bs on Orion.

BinLiu-NOAA committed:
Load the module g2/3.1.1 on Orion

BinLiu-NOAA committed:
Merge branch 'support/HAFS' into feature/multi_nests

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/origin/support/HAFS' into feature/multi_nests

BinLiu-NOAA committed:
Merge branch 'feature/multi_nests' of https://github.com/hafs-community/UFS_UTILS into feature/multi_nests

BinLiu-NOAA committed:
Merge branch 'feature/multi_nests' of https://github.com/hafs-community/UFS_UTILS into feature/multi_nests

BinLiu-NOAA committed:
Fix a typo for parent_tile in sorc/fre-nctools.fd/tools/make_hgrid/create_gnomonic_cubic_grid.c (Bill and Bin).

BinLiu-NOAA committed:
Bring in the latest master version of make_hgrid from NOAA-GFDL/FRE-NCtools (as
of 04/02/2021), which has the capability of generating gnomonic cubic grids for
multiple and telescopic nested domains.

BinLiu-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'support/HAFS' into feature/multi_nests

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/origin/support/HAFS' into feature/multi_nests

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

GeorgeGayno-NOAA committed:
Update machine-setup.sh for wcoss2. Baseline initial
build module for that machine.

Fixes #559.

GeorgeGayno-NOAA committed:
Kyle's updates for finding the wgrib2 library.

Fixes #559

GeorgeGayno-NOAA committed:
Update link_fixdirs.sh for wcoss2. For now, Kate will
host the fixed files in her own directory.

Fixes #559.

GeorgeGayno-NOAA committed:
Load additional libraries from the stack instead of letting
cmake find another version.

Fixes #559

GeorgeGayno-NOAA committed:
Add grid generation driver script for WCOSS2.

Fixes #559.

GeorgeGayno-NOAA committed:
Update vcoord_gen utility for wcoss2.

Fixes #559

GeorgeGayno-NOAA committed:
Create ice_blend regression test driver script for wcoss2.

Fixes #559.

GeorgeGayno-NOAA committed:
Create sfc_climo_gen utility script for wcoss2.

Fixes #559.

GeorgeGayno-NOAA committed:
Add snow2mdl regression test script for wcoss2.

Fixes #559.

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Fix a fprintf line bug when printing parent_tile.

BinLiu-NOAA committed:
Sync make_hgrid with the latest master branch version in NOAA-GFDL/FRE-NCtools.

GeorgeGayno-NOAA committed:
Add global_cycle regression test script for wcoss2.

Fixes #559.

BinLiu-NOAA committed:
Sync make_solo_mosaic with the latest master branch version (as of 202109) in NOAA-GFDL/FRE-NCtools.

BinLiu-NOAA committed:
Sync fregrid with the latest master branch version (as of 202109) in NOAA-GFDL/FRE-NCtools.
Also synced fre-nctools.fd/shared_lib/mpp_io.c/h.

BinLiu-NOAA committed:
Merge branch 'feature/multi_nests' of https://github.com/hafs-community/UFS_UTILS into feature/multi_nests

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/origin/support/HAFS' into feature/multi_nests

GeorgeGayno-NOAA committed:
Add grid_gen regression test driver script for wcoss2.

Fixes #559.

GeorgeGayno-NOAA committed:
Minor updates to global_cycle reg test script.

Fixes #559

GeorgeGayno-NOAA committed:
Add chgres_cube regression test driver script for wcoss2.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Begin work on new driver script for gdas_init utility.

Fixes #559.

GeorgeGayno-NOAA committed:
Complete mods for data pull scripts. Start mods for chgres
scripts.

Fixes #559.

GeorgeGayno-NOAA committed:
Add qsub commands to remainder of ./util/gdas_init/driver.wcoss2.sh
script.

Fixes #559

GeorgeGayno-NOAA committed:
Hardware threading was turned off on WCOSS2. Start adjusting
the scripts accordingly.

Fixes #559.

GeorgeGayno-NOAA committed:
Update more scripts now that hardware threading is
turned off.

Fixes #559.

GeorgeGayno-NOAA committed:
Update more scripts for new hardware configuration.

Fixes #559.

GeorgeGayno-NOAA committed:
Update gdas_init driver script for new hardware
configuration.

Fixes #559.

GeorgeGayno-NOAA committed:
Update ./util/gdas_init/driver.wcoss2.sh to process all
80 enkf members.

Fixes #559

GeorgeGayno-NOAA committed:
Update ice blend test to use v1.2.3 of grib_util. v1.2.2
gave wrong results when replacing the bitmap with the 1.57
flag value.

Fixes #559.

GeorgeGayno-NOAA committed:
Update build module to point to latest esmf v8.

Fixes #559

GeorgeGayno-NOAA committed:
Update consistency test driver scripts to point
to my local copy of 'nccmp'. Hope this is temporary.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Update fix/link_fixdirs.sh to point to the correct 'fix'
directories.

Fixes #559.

GeorgeGayno-NOAA committed:
Point to new location for data and utilities on wcoss2.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Remove hard-wired library versions per NCO directive for
WCOSS2.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/origin/support/HAFS' into feature/multi_nests

GeorgeGayno-NOAA committed:
Change references 'wcoss2_cray' to 'wcoss2'.

Fixes #559.

GeorgeGayno-NOAA committed:
Update wcoss2 build module to be lua compliant.

Fixes #559.

GeorgeGayno-NOAA committed:
More updates to wcoss2 build process for lua compliance.

Fixes #559.

GeorgeGayno-NOAA committed:
Update utility scripts for new lua based build module.

Fixes #559.

GeorgeGayno-NOAA committed:
Update regression test scripts for new lua-based build
module.

Fixes #559.

GeorgeGayno-NOAA committed:
Update grid creation driver script for new lua-compliant
build module.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Remove all references to jlogfile from scripts per NCO
requirements.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/wcoss2

Fixes #559

GeorgeGayno-NOAA committed:
Update library version defaults to match gfs v16.2.

Remove load of envvar and replace with 'module reset'.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559

BinLiu-NOAA committed:
Switch to load esmf/8_2_0 on hera.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559

GeorgeGayno-NOAA committed:
Update wcoss2 regression test scripts for new baseline
update logic.

Fixes #559.

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into support/HAFS

BinLiu-NOAA committed:
Merge branch 'support/HAFS' into feature/multi_nests

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

BijuThomas-NOAA committed:
Adding mppnccombine to hafs_utils.fd

BinLiu-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/hafsv0.3_baseline

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Increase memory for some chgres grib2 consistency tests.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into feature/hafsv0.3_baseline

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Minor updates to ./reg_tests/chgres_cube/driver.wcoss2.sn

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Minor update to build script.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

GeorgeGayno-NOAA committed:
Remove load of wgrib2 from build module.

Fixes #559.

GeorgeGayno-NOAA committed:
Minor script updates.

Fixes #559.

GeorgeGayno-NOAA committed:
Update ./reg_tests/rt.sh for wcoss2.

Fixes #559.

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/wcoss2

Fixes #559.

BinLiu-NOAA committed:
Merge commit 'b23a25a218deb20113b6cf18002525df2e06841d' into feature/hafsv0.3_final.
Merge the 05/03/2022 version of develop branch into the feature/hafsv0.3_final branch.

GeorgeGayno-NOAA committed:
Update rt.sh to check for which wcoss2 is production and
development.

Fixes #559.

Commit:
Merge remote-tracking branch 'remotes/GG/feature/wcoss2' into feature/hafsv0.3_final

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into feature/hafs_sync_202210

BinLiu-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/hafs_sync_202210

BinLiu-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/hafs_sync_202210

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/upstream/develop' into feature/hafs.v1.0.0

BinLiu-NOAA committed:
Move mppnccombine from sorc/fre-nctools.fd/tools to sorc/fre-nctools.fd/postprocessing.

BinLiu-NOAA committed:
Remove duplicated wcoss2 related changes in reg_tests/rt.sh.

GeorgeGayno-NOAA reviewed (changes requested):
Any new codes, such as mppnccombine, will need a test.

Any new functions to existing codes will also need a test. What exactly changed with 'fregrid', 'make_hgrid' and 'make_solo_mosaic'?

GeorgeGayno-NOAA commented:
I don't see 'develop' in your fork. What exactly are you branching from?

BinLiu-NOAA commented:
> Any new codes, such as mppnccombine, will need a test.
> 
> Any new functions to existing codes will also need a test. What exactly changed with 'fregrid', 'make_hgrid' and 'make_solo_mosaic'?

@GeorgeGayno-NOAA, as for mppnccombine, I would like to check with @JacobCarley-NOAA and @JianpingHuang-NOAA to see if they actually still use/need this mppnccombine tool. For HAFS, we previously used it as a workaround for a slow IO issue for FV3 restart files. But, we currently no longer use it as the slow IO for FV3 restart files has been addressed from the ufs-weather-model/FV3 side.

In terms of the fregrid, make_hgrid, make_solo_mosaic, and other fre-nctools.fd related changes, this set of changes are mainly a manual sync with the corresponding version from [FRE-NCtools official repository](https://github.com/NOAA-GFDL/FRE-NCtools) as of 09/2021 to bring in the support for multiple static and moving nesting capabilities needed by HAFS.

BinLiu-NOAA commented:
> I don't see 'develop' in your fork. What exactly are you branching from?

@GeorgeGayno-NOAA, this branch was originally branched out from UFS_UTILS master branch a while ago (at least 2 years or more). But, we have been keeping syncing with the UFS_UTILS's develop/master branch. And you can see the actual differences between this branch and the latest develop branch are relatively small (only 20+ files were changed/added). Also, if you would prefer to, all these changes can just be squashed and merged back on top of the latest develop branch, once the PR review and test are done. Thanks!

BinLiu-NOAA committed:
Merge branch 'develop' into feature/hafs.v1.0.0

JianpingHuang-NOAA commented:
@Bin Liu - NOAA Affiliate ***@***.***>   We used to
use mppnccombine utility to support UFS-AQM testing at the beginning. It is
fine for us without using it now.

On Wed, Jan 4, 2023 at 8:53 AM Bin Liu ***@***.***> wrote:

> Any new codes, such as mppnccombine, will need a test.
>
> Any new functions to existing codes will also need a test. What exactly
> changed with 'fregrid', 'make_hgrid' and 'make_solo_mosaic'?
>
> @GeorgeGayno-NOAA <https://github.com/GeorgeGayno-NOAA>, as for
> mppnccombine, I would like to check with @JacobCarley-NOAA
> <https://github.com/JacobCarley-NOAA> and @JianpingHuang-NOAA
> <https://github.com/JianpingHuang-NOAA> to see if they actually still
> use/need this mppnccombine tool. For HAFS, we previously used it as a
> workaround for a slow IO issue for FV3 restart files. But, we currently no
> longer use it as the slow IO for FV3 restart files has been addressed from
> the ufs-weather-model/FV3 side.
>
> In terms of the fregrid, make_hgrid, make_solo_mosaic, and other
> fre-nctools.fd related changes, this set of changes are mainly a manual
> sync with the corresponding version from FRE-NCtools official repository
> <https://github.com/NOAA-GFDL/FRE-NCtools> as of 09/2021 to bring in the
> support for multiple static and moving nesting capabilities needed by HAFS.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/UFS_UTILS/pull/752#issuecomment-1370955790>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANA2PI5O5ACIWRILO2VGPOLWQV6D7ANCNFSM6AAAAAATLYXV5M>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

BinLiu-NOAA commented:
> @bin Liu - NOAA Affiliate ***@***.***> We used to use mppnccombine utility to support UFS-AQM testing at the beginning. It is fine for us without using it now.
> […](#)
> On Wed, Jan 4, 2023 at 8:53 AM Bin Liu ***@***.***> wrote: Any new codes, such as mppnccombine, will need a test. Any new functions to existing codes will also need a test. What exactly changed with 'fregrid', 'make_hgrid' and 'make_solo_mosaic'? @GeorgeGayno-NOAA <https://github.com/GeorgeGayno-NOAA>, as for mppnccombine, I would like to check with @JacobCarley-NOAA <https://github.com/JacobCarley-NOAA> and @JianpingHuang-NOAA <https://github.com/JianpingHuang-NOAA> to see if they actually still use/need this mppnccombine tool. For HAFS, we previously used it as a workaround for a slow IO issue for FV3 restart files. But, we currently no longer use it as the slow IO for FV3 restart files has been addressed from the ufs-weather-model/FV3 side. In terms of the fregrid, make_hgrid, make_solo_mosaic, and other fre-nctools.fd related changes, this set of changes are mainly a manual sync with the corresponding version from FRE-NCtools official repository <https://github.com/NOAA-GFDL/FRE-NCtools> as of 09/2021 to bring in the support for multiple static and moving nesting capabilities needed by HAFS. — Reply to this email directly, view it on GitHub <[#752 (comment)](https://github.com/ufs-community/UFS_UTILS/pull/752#issuecomment-1370955790)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ANA2PI5O5ACIWRILO2VGPOLWQV6D7ANCNFSM6AAAAAATLYXV5M> . You are receiving this because you were mentioned.Message ID: ***@***.***>

@JianpingHuang-NOAA, thanks for confirming you are no longer using this mppnccombine tool! @GeorgeGayno-NOAA  I am trying to check with the RRFS group. If they also do not use it. I will just go ahead take out the mppnccombine changes. Thanks!

GeorgeGayno-NOAA commented on exec/.gitignore:
```diff
@@ -0,0 +1,7 @@
+# git does not allow empty directories.
```
Why do you need the 'exec' directory to exist? It is automatically created during the cmake build.

BinLiu-NOAA replied:
@GeorgeGayno-NOAA, The only reason we add this empty exec dir in the repository is that one can know the directory structure (exec dir) before building the package. But, anyway, I have deleted this exec/.gitignore file in this PR. 

P.S., another advantage of having this exec/.gitignore file is to prevent someone incidentally adding/committing those exec/executable.x files back to the repository.

GeorgeGayno-NOAA commented:
> > @bin Liu - NOAA Affiliate _**@**_._**> We used to use mppnccombine utility to support UFS-AQM testing at the beginning. It is fine for us without using it now.
> > […](#)
> > On Wed, Jan 4, 2023 at 8:53 AM Bin Liu _**@**_.**_> wrote: Any new codes, such as mppnccombine, will need a test. Any new functions to existing codes will also need a test. What exactly changed with 'fregrid', 'make_hgrid' and 'make_solo_mosaic'? @GeorgeGayno-NOAA https://github.com/GeorgeGayno-NOAA, as for mppnccombine, I would like to check with @JacobCarley-NOAA https://github.com/JacobCarley-NOAA and @JianpingHuang-NOAA https://github.com/JianpingHuang-NOAA to see if they actually still use/need this mppnccombine tool. For HAFS, we previously used it as a workaround for a slow IO issue for FV3 restart files. But, we currently no longer use it as the slow IO for FV3 restart files has been addressed from the ufs-weather-model/FV3 side. In terms of the fregrid, make_hgrid, make_solo_mosaic, and other fre-nctools.fd related changes, this set of changes are mainly a manual sync with the corresponding version from FRE-NCtools official repository https://github.com/NOAA-GFDL/FRE-NCtools as of 09/2021 to bring in the support for multiple static and moving nesting capabilities needed by HAFS. — Reply to this email directly, view it on GitHub <[#752 (comment)](https://github.com/ufs-community/UFS_UTILS/pull/752#issuecomment-1370955790)>, or unsubscribe https://github.com/notifications/unsubscribe-auth/ANA2PI5O5ACIWRILO2VGPOLWQV6D7ANCNFSM6AAAAAATLYXV5M . You are receiving this because you were mentioned.Message ID: _**@**_.***>
> 
> @JianpingHuang-NOAA, thanks for confirming you are no longer using this mppnccombine tool! @GeorgeGayno-NOAA I am trying to check with the RRFS group. If they also do not use it. I will just go ahead take out the mppnccombine changes. Thanks!

Thank you for not adding unneeded programs to the repository.

GeorgeGayno-NOAA commented:
> One of the `grid_gen` consistency tests failed. More details to follow.

The test that checks a GFDL regional grid (See https://github.com/ufs-community/UFS_UTILS/blob/develop/reg_tests/grid_gen/gfdl.regional.sh) failed. Differences in the 'grid' file are small, however:

```
+ /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-2022.1.2/nccmp/1.8.9.0/bin/nccmp -dmfqS C424_grid.tile7.halo4.nc /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/gfdl.regional/C424_grid.tile7.halo4.nc
Variable Group  Count          Sum      AbsSum          Min         Max       Range         Mean      StdDev
x        /      36678  5.20686e-11  2.2697e-09 -6.25278e-13 6.82121e-13  1.3074e-12  1.41961e-15 6.72792e-14
y        /     110714 -9.49463e-12 1.60327e-09 -5.11591e-13 5.18696e-13 1.03029e-12 -8.57581e-17 2.10389e-14
dx       /     122195  5.39039e-08 0.000152321 -1.39898e-08 1.40153e-08 2.80052e-08   4.4113e-13 2.38109e-09
dy       /     117889  5.88534e-08 0.000108247 -1.06102e-08 1.06102e-08 2.12203e-08  4.99227e-13 1.15666e-09
area     /     123698       14.259      2304.8   -0.0721062   0.0721062    0.144212  0.000115273   0.0213371
```

This lead to downstream changes in the 'orog' files (which are also small)

```
+ /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-2022.1.2/nccmp/1.8.9.0/bin/nccmp -dmfqS C424_oro_data.tile7.halo0.nc /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/gfdl.regional/C424_oro_data.tile7.halo0.nc
Variable  Group Count         Sum     AbsSum          Min         Max      Range         Mean      StdDev
land_frac /        63 -0.00283378  0.0270154  -0.00405771  0.00222635 0.00628406 -4.49806e-05 0.000857066
orog_raw  /       155     2.68667    22.3095     -1.18295     2.52661    3.70956    0.0173334    0.372257
orog_filt /      1283     2.69182    26.5098     -1.32559     1.12305    2.44864   0.00209807    0.089028
stddev    /       157     3.00501    15.0479     -1.66577     1.83842     3.5042    0.0191402    0.297264
convexity /       143   -0.202715    1.21825    -0.113843   0.0579295   0.171773  -0.00141759   0.0158465
theta     /       143    -1.95116     42.125     -3.79125     2.44946    6.24071   -0.0136445    0.597306
gamma     /       143  -0.0226614   0.247327   -0.0134753  0.00776458  0.0212399 -0.000158472  0.00276828
sigma     /       143 5.28634e-05 0.00597061 -0.000411563 0.000698715 0.00111028  3.69674e-07 0.000115382
elvmax    /       156    -2.68683    22.3097     -2.52649     1.18292    3.70941   -0.0172232    0.371053
```

BinLiu-NOAA committed:
Remove mppnccombine tool, which is no longer needed for HAFS and other applications.

BinLiu-NOAA committed:
Remove exec/.gitignore, which is not absolutely needed and the exec dir will be genearated during the cmake build anyway.

BinLiu-NOAA commented:
> Thank you for not adding unneeded programs to the repository.

@GeorgeGayno-NOAA, I have checked with developers for other applications (RRFS, AQM), and they no longer use this mppnccombine tool. So I have just removed it from this PR. PR title and description are also updated now. Thanks!

BinLiu-NOAA commented:
> > One of the `grid_gen` consistency tests failed. More details to follow.

@GeorgeGayno-NOAA, I think these differences are due to the sync with the fre-nctool official repository, which probably included some bug fixes/updates. I hope this result changing can be considered as acceptable. Thanks!

GeorgeGayno-NOAA commented:
> > > One of the `grid_gen` consistency tests failed. More details to follow.
> 
> @GeorgeGayno-NOAA, I think these differences are due to the sync with the fre-nctools official repository, which probably included some bug fixes/updates. I hope this result changing can be considered as acceptable. Thanks!

These differences are very small and will not prevent a merge. I am not seeing any differences with global uniform grids. (I tried C96 and C768).

GeorgeGayno-NOAA commented:
Does this merge support a HAFS implementation? If so, will you need a UFS_UTILS tag or will you create a tag from the HAFS fork?

BinLiu-NOAA commented:
> Does this merge support a HAFS implementation? If so, will you need a UFS_UTILS tag or will you create a tag from the HAFS fork?

@GeorgeGayno-NOAA, yes. I possible, we would prefer your help to create a support/production branch (say production/hafs.v1.0.0 branch or support/hafs.v1.0.0 branch depending upon ufs_utils side branch naming convention) from the develop branch once this PR is merged. This is just in case any further changes might be needed before the HAFSv1 code freeze and code delivery to NCO. And for the NCO code delivery, you can then help to tag this branch into a tag like ops-hafsv1.0.0 (or simply hafs.v1.0.0 or something similar). Please feel free to let us know if you would prefer to something else. Thanks!

GeorgeGayno-NOAA commented:
> > Does this merge support a HAFS implementation? If so, will you need a UFS_UTILS tag or will you create a tag from the HAFS fork?
> 
> @GeorgeGayno-NOAA, yes. If possible, we would prefer your help to create a support/production branch (say production/hafs.v1.0.0 branch or support/hafs.v1.0.0 branch depending upon ufs_utils side branch naming convention) from the develop branch once this PR is merged. This is just in case any further changes might be needed before the HAFSv1 code freeze and code delivery to NCO. And for the NCO code delivery, you can then help to tag this branch into a tag like ops-hafsv1.0.0 (or simply hafs.v1.0.0 or something similar). Please feel free to let us know if you would prefer to something else. Thanks!

We have been asked to do versioned tags - v1.8 etc. So, after the merge we can create v1.9. If further changes are required, we can create another tag - v1.10 - or if the fix is small, add it to v1.9 and then create a v1.9.1 tag.

BinLiu-NOAA commented:
> We have been asked to do versioned tags - v1.8 etc. So, after the merge we can create v1.9. If further changes are required, we can create another tag - v1.10 - or if the fix is small, add it to v1.9 and then create a v1.9.1 tag.

@GeorgeGayno-NOAA, I saw there are tags like [ops-gfsv16.3.0](https://github.com/ufs-community/UFS_UTILS/releases/tag/ops-gfsv16.3.0). So, this tag naming convention is no longer recommended? If not, we can use whatever the UFS_UTILS repository side preferred tag/branch names. Thanks!

GeorgeGayno-NOAA commented:
> > We have been asked to do versioned tags - v1.8 etc. So, after the merge we can create v1.9. If further changes are required, we can create another tag - v1.10 - or if the fix is small, add it to v1.9 and then create a v1.9.1 tag.
> 
> @GeorgeGayno-NOAA, I saw there are tags like [ops-gfsv16.3.0](https://github.com/ufs-community/UFS_UTILS/releases/tag/ops-gfsv16.3.0). So, this tag naming convention is no longer recommended? If not, we can use whatever the UFS_UTILS repository side preferred tag/branch names. Thanks!

We were told to stop naming tags after the specific operational implementation. You are seeing some tags with the old naming convention that we need to maintain.

BinLiu-NOAA commented:
@GeorgeGayno-NOAA, Thanks for the context and explanation! And we can go with the UFS_UTILS repository preferred naming convention for tagging/branching. Thanks!

BinLiu-NOAA commented:
@GeorgeGayno-NOAA, wondering if it is possible to merge this PR back to develop branch and create a tag for HAFSv1 code freeze saying before the end of this week. Much appreciated.

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA, wondering if it is possible to merge this PR back to develop branch and create a tag for HAFSv1 code freeze saying before the end of this week. Much appreciated.

Yes, I can create your tag before the end of this week.

GeorgeGayno-NOAA commented on sorc/machine-setup.sh:
```diff
@@ -19,7 +19,7 @@ fi
 target=""
 USERNAME=`echo $LOGNAME | awk '{ print tolower($0)'}`
 
```
Just curious why this was changed.

GeorgeGayno-NOAA commented:
> > One of the `grid_gen` consistency tests failed. More details to follow.
> 
> The test that checks a GFDL regional grid (See https://github.com/ufs-community/UFS_UTILS/blob/develop/reg_tests/grid_gen/gfdl.regional.sh) failed. Differences in the 'grid' file are small, however:
> 
> ```
> + /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-2022.1.2/nccmp/1.8.9.0/bin/nccmp -dmfqS C424_grid.tile7.halo4.nc /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/gfdl.regional/C424_grid.tile7.halo4.nc
> Variable Group  Count          Sum      AbsSum          Min         Max       Range         Mean      StdDev
> x        /      36678  5.20686e-11  2.2697e-09 -6.25278e-13 6.82121e-13  1.3074e-12  1.41961e-15 6.72792e-14
> y        /     110714 -9.49463e-12 1.60327e-09 -5.11591e-13 5.18696e-13 1.03029e-12 -8.57581e-17 2.10389e-14
> dx       /     122195  5.39039e-08 0.000152321 -1.39898e-08 1.40153e-08 2.80052e-08   4.4113e-13 2.38109e-09
> dy       /     117889  5.88534e-08 0.000108247 -1.06102e-08 1.06102e-08 2.12203e-08  4.99227e-13 1.15666e-09
> area     /     123698       14.259      2304.8   -0.0721062   0.0721062    0.144212  0.000115273   0.0213371
> ```
> 
> This lead to downstream changes in the 'orog' files (which are also small)
> 
> ```
> + /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-2022.1.2/nccmp/1.8.9.0/bin/nccmp -dmfqS C424_oro_data.tile7.halo0.nc /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/gfdl.regional/C424_oro_data.tile7.halo0.nc
> Variable  Group Count         Sum     AbsSum          Min         Max      Range         Mean      StdDev
> land_frac /        63 -0.00283378  0.0270154  -0.00405771  0.00222635 0.00628406 -4.49806e-05 0.000857066
> orog_raw  /       155     2.68667    22.3095     -1.18295     2.52661    3.70956    0.0173334    0.372257
> orog_filt /      1283     2.69182    26.5098     -1.32559     1.12305    2.44864   0.00209807    0.089028
> stddev    /       157     3.00501    15.0479     -1.66577     1.83842     3.5042    0.0191402    0.297264
> convexity /       143   -0.202715    1.21825    -0.113843   0.0579295   0.171773  -0.00141759   0.0158465
> theta     /       143    -1.95116     42.125     -3.79125     2.44946    6.24071   -0.0136445    0.597306
> gamma     /       143  -0.0226614   0.247327   -0.0134753  0.00776458  0.0212399 -0.000158472  0.00276828
> sigma     /       143 5.28634e-05 0.00597061 -0.000411563 0.000698715 0.00111028  3.69674e-07 0.000115382
> elvmax    /       156    -2.68683    22.3097     -2.52649     1.18292    3.70941   -0.0172232    0.371053
> ```

The branch at af65b7e was compiled on Hera, Jet, Orion and Cactus. The grid_gen consistency test were run. All test passed expect the GFDL regional test. It failed on all machines with the very small errors as described above. I can't explain these differences, but they are too small to worry about.

GeorgeGayno-NOAA commented on sorc/machine-setup.sh:
```diff
@@ -19,7 +19,7 @@ fi
 USERNAME=`echo $LOGNAME | awk '{ print tolower($0)'}`
 
-if [[ -d /lfs3 ]] ; then
+if [[ -d /lfs4 ]] ; then
```
Just curious why the directory was changed here.

BinLiu-NOAA replied:
@GeorgeGayno-NOAA I thought /lfs3 disk is no longer available on Jet. Only /lfs1 and /lfs4.

GeorgeGayno-NOAA replied:
@BinLiu-NOAA You are correct. The /lfs3 directory is empty, but machine-setup.sh was still working. Thanks for catching this.

GeorgeGayno-NOAA reviewed (approved):
Looks good.
